indico/web/templates/forms/principal_list_widget.html

```diff
@@ -25,12 +25,20 @@
 
             function addPrincipal(newPrincipals, setResult) {
                 // remove existing ones first to avoid duplicates
-                _.each(newPrincipals, function(principal) {
-                    principals = _.without(principals, _.findWhere(principals, {
-                        identifier: principal.identifier
-                    }));
+                newPrincipals = _.filter(newPrincipals, function(principal) {
+                    return !_.findWhere(principals, {identifier: principal.identifier});
+                });
+                principals = _.sortBy(principals.concat(newPrincipals), function(principal) {
+                    var weight;
+                    if (principal._type == 'Avatar') {
+                        weight = 0;
+                    } else if (principal.isGroup) {
+                        weight = 1;
+                    } else {
+                        weight = 2;
+                    }
```

Comment:
If you decide to use the one-liner I mentioned before, you should use it here too.